src/dbnode/persist/fs/commitlog/writer.go

```diff
@@ -203,34 +197,13 @@ func (w *writer) Write(
 
 	seen := w.seen.Test(uint(series.UniqueIndex))
 	if !seen {
-		var encodedTags []byte
-		if series.EncodedTags != nil {
-			// If already serialized use the serialized tags.
-			encodedTags = series.EncodedTags
-		} else if series.Tags.Values() != nil {
-			// Otherwise serialize the tags.
-			w.tagSliceIter.Reset(series.Tags)
-			w.tagEncoder.Reset()
-			err := w.tagEncoder.Encode(w.tagSliceIter)
-			if err != nil {
-				return err
-			}
-
-			encodedTagsChecked, ok := w.tagEncoder.Data()
-			if !ok {
-				return errTagEncoderDataNotAvailable
-			}
-
-			encodedTags = encodedTagsChecked.Bytes()
-		}
-
 		// If "idx" likely hasn't been written to commit log
 		// yet we need to include series metadata
 		var metadata schema.LogMetadata
 		metadata.ID = series.ID.Bytes()
 		metadata.Namespace = series.Namespace.Bytes()
 		metadata.Shard = series.Shard
-		metadata.EncodedTags = encodedTags
+		metadata.EncodedTags = series.EncodedTags
```

Comment:
Will `series.EncodingTags` always be non nil now? Looks like the old path would have put in an empty byte slice if there weren't tags, but may be nil now